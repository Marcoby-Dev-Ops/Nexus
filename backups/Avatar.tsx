import React from 'react';
import PropTypes from 'prop-types';

/**
 * @interface AvatarProps
 * @description Props for the Avatar component.
 */
export interface AvatarProps {
  src?: string;
  alt?: string;
  initials?: string;
  className?: string;
}

/**
 * @name Avatar
 * @description A user avatar component with fallback to initials.
 * @param {AvatarProps} props - The props for the component.
 * @returns {JSX.Element} The rendered Avatar component.
 */
export const Avatar: React.FC<AvatarProps> = React.memo(({ src, alt = '', initials, className = '' }) => (
  <div className={`inline-flex items-center justify-center rounded-full bg-muted text-foreground w-10 h-10 ${className}`}> 
    {src ? (
      <img src={src} alt={alt} className="w-full h-full object-cover rounded-full" />
    ) : (
      <span className="font-semibold">{initials || '?'}</span>
    )}
  </div>
));

Avatar.displayName = 'Avatar';

Avatar.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  initials: PropTypes.string,
  className: PropTypes.string,
};

export default Avatar; 