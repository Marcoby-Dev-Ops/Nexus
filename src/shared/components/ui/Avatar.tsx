import React from 'react';
import { cn } from '@/shared/utils/styles';
import PropTypes from 'prop-types';

/**
 * @interface AvatarProps
 * @description Props for the Avatar component.
 */
export interface AvatarProps extends React.HTMLAttributes<HTMLSpanElement> {}

/**
 * @interface AvatarImageProps
 * @description Props for the AvatarImage component.
 */
export interface AvatarImageProps extends React.ImgHTMLAttributes<HTMLImageElement> {}

/**
 * @interface AvatarFallbackProps
 * @description Props for the AvatarFallback component.
 */
export interface AvatarFallbackProps extends React.HTMLAttributes<HTMLSpanElement> {}

/**
 * @name Avatar
 * @description A container for avatar content with proper styling.
 * @param {AvatarProps} props - The props for the component.
 * @returns {JSX.Element} The rendered Avatar component.
 */
export const Avatar: React.FC<AvatarProps> = React.memo(({ 
  className, 
  ...props 
}) => (
  <span
    className={cn(
      'relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full',
      className
    )}
    {...props}
  />
));

/**
 * @name AvatarImage
 * @description An image component for avatars with proper styling.
 * @param {AvatarImageProps} props - The props for the component.
 * @returns {JSX.Element} The rendered AvatarImage component.
 */
export const AvatarImage: React.FC<AvatarImageProps> = React.memo(({ 
  className, 
  ...props 
}) => (
  <img
    className={cn('aspect-square h-full w-full', className)}
    {...props}
  />
));

/**
 * @name AvatarFallback
 * @description A fallback component for avatars when image is not available.
 * @param {AvatarFallbackProps} props - The props for the component.
 * @returns {JSX.Element} The rendered AvatarFallback component.
 */
export const AvatarFallback: React.FC<AvatarFallbackProps> = React.memo(({ 
  className, 
  ...props 
}) => (
  <span
    className={cn(
      'flex h-full w-full items-center justify-center rounded-full bg-muted text-foreground font-medium',
      className
    )}
    {...props}
  />
));

Avatar.displayName = 'Avatar';
AvatarImage.displayName = 'AvatarImage';
AvatarFallback.displayName = 'AvatarFallback';

Avatar.propTypes = {
  className: PropTypes.string,
};

AvatarImage.propTypes = {
  className: PropTypes.string,
  src: PropTypes.string,
  alt: PropTypes.string,
};

AvatarFallback.propTypes = {
  className: PropTypes.string,
};

export default Avatar; 