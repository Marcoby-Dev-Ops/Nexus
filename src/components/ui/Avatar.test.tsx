/**
 * @file Avatar.test.tsx
 * @description Unit and snapshot tests for the Avatar component.
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Avatar from './Avatar';

describe('Avatar', () => {
  it('renders with alt text', () => {
    render(<Avatar src="/avatar.png" alt="User Avatar" />);
    expect(screen.getByAltText('User Avatar')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Avatar src="/avatar.png" alt="User Avatar" />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 