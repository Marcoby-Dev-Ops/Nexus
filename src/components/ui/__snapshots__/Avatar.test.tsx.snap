// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="inline-flex items-center justify-center rounded-full bg-muted text-foreground w-10 h-10 "
  >
    <img
      alt="User Avatar"
      class="w-full h-full object-cover rounded-full"
      src="/avatar.png"
    />
  </div>
</DocumentFragment>
`;
